spk/pyload/src/installer.sh

```diff
@@ -0,0 +1,146 @@
+#!/bin/sh
+
+# Package
+PACKAGE="pyload"
+DNAME="pyLoad"
+
+# Others
+INSTALL_DIR="/usr/local/${PACKAGE}"
+PATH="${INSTALL_DIR}/bin:/usr/local/bin:/bin:/usr/bin:/usr/syno/bin:/usr/syno/sbin"
+SERVICETOOL="/usr/syno/bin/servicetool"
+FWPORTS="/var/packages/${PACKAGE}/scripts/${PACKAGE}.sc"
+
+lng2iso()
+{
+	# changes 3-character Synology language code to ISO 639-1 code.
+        case $1 in
+            eng|deu|fre|ita|nld|sve|rus|plk|csy)
+                echo "${1%?}"
+                ;;
+            spn)
+                echo "es"
+                ;;
+            *)
+		echo "en"
+		;;
+        esac
+}
+
+
+preinst ()
+{
+    if [ "${SYNOPKG_PKG_STATUS}" == "INSTALL" ]; then
+        test "${wizard_download_dir}" || wizard_download_dir="/volume1/downloads"
+
+        mkdir -p "${SYNOPKG_PKGINST_TEMP_DIR}/etc"
+        touch "${SYNOPKG_PKGINST_TEMP_DIR}/etc/pyload-pkg.conf"
+
+        # Check whether ${wizard_download_dir} is in existing shared folder.
+        VOLUME=$(echo "${wizard_download_dir}" | cut -d/ -f2)
+        WIZ_SHARE=$(echo "${wizard_download_dir}" | cut -d/ -f3)
+        SHARE_EXISTS=$(synoshare --get "${WIZ_SHARE}" | sed -ne 's/^[[:space:]]*Path[[:space:]\.]*\[\(.*\)\].*$/\1/p')  # FIXME: Better variable name
+
+        if [ -z "${SHARE_EXISTS}" ]; then
+            synoshare --add "${WIZ_SHARE}" "" "/${VOLUME}/${WIZ_SHARE}" "" "" "" 1 0 > ${SYNOPKG_TEMP_LOGFILE} || exit 1
```

Comment:
I'm not a fan of creating shares in packages, just test if the folder the user input exists and if not just exit gracefully with an error message.